ktor-client/ktor-client-features/ktor-client-auth/common/src/io/ktor/client/features/auth/providers/DigestAuthProvider.kt

```diff
@@ -28,31 +29,91 @@ public fun Auth.digest(block: DigestAuthConfig.() -> Unit) {
  */
 @Suppress("KDocMissingDocumentation")
 public class DigestAuthConfig {
-    public var username: String = ""
-    public var password: String = ""
-    public var realm: String? = null
+
     public var algorithmName: String = "MD5"
+
+    /**
+     * Required: The username of the basic auth.
+     */
+    @Deprecated("Please use `credentials {}` function instead")
+    public lateinit var username: String
```

Comment:
Breaking change with lateinit. It will be an exception if the user doesn't configure the feature.